### PR TITLE
Use platform to fetch user messages

### LINKED ISSUE #202: Use platform to fetch user messages
```
# TODO: Move input part to Platform.
text = input("Your response: ")
```

### PR DESCRIPTION
**Pull request by IKostric**
Fixes #202

### CODE REVIEW

File: dialoguekit/platforms/platform.py [hunk 1]
```diff
@@ -1,25 +1,47 @@
...
     def __init__(self):
         """Represents a platform."""
-        pass
+        self._user_callback: Callable[[str], None] = None
+
+    def register_user_callback(self, callback: Callable[[str], None]) -> None:
+        """Registers a callback function to be called on user input.
+
+        Args:
+            callback: Function to call on user input
```
Comment (NoB0) [inline dialoguekit/platforms/platform.py:18]:
```suggestion
            callback: Function to call on user input.
```

File: dialoguekit/platforms/platform.py [hunk 2]
```diff
@@ -1,25 +1,47 @@
...
+        """
+        self._user_callback = callback
+
+    @abstractmethod
+    def listen_for_user_input(self) -> None:
+        """Listens for the user input.
+
+        Args:
+            callback: Function to call on user input
```
Comment (NoB0) [inline dialoguekit/platforms/platform.py:27]:
```suggestion
            callback: Function to call on user input.
```

File: dialoguekit/platforms/platform.py [hunk 3]
```diff
@@ -1,25 +1,47 @@
...
+        """Registers a callback function to be called on user input.
+
+        Args:
+            callback: Function to call on user input
+        """
+        self._user_callback = callback
+
+    @abstractmethod
+    def listen_for_user_input(self) -> None:
+        """Listens for the user input.
```
Comment (NoB0) [inline dialoguekit/platforms/platform.py:24]: Nit: Maybe a Raises note can be added to doctrings.

File: dialoguekit/platforms/platform.py [hunk 4]
```diff
@@ -1,25 +1,47 @@
...
 
         Args:
             utterance: An instance of Utterance.
         """
-        print(f"AGENT: {utterance.text}")
+        raise NotImplementedError
 
+    @abstractmethod
     def display_user_utterance(self, utterance: Utterance) -> None:
         """Diplays a user utterance.
```
Comment (NoB0) [inline dialoguekit/platforms/platform.py:42]:
```suggestion
        """Displays a user utterance.
```

File: dialoguekit/platforms/platform.py [hunk 5]
```diff
@@ -1,25 +1,47 @@
...
 
         Args:
             utterance: An instance of Utterance.
         """
```
Comment (NoB0) [inline dialoguekit/platforms/platform.py:46]: Nit: Maybe a Raises note can be added to doctrings.

File: dialoguekit/platforms/platform_terminal.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
+"""The Platform facilitates displaying of the conversation."""
```
Comment (NoB0) [inline dialoguekit/platforms/platform_terminal.py:1]: Nit: maybe you can mention here that it is the terminal.

File: dialoguekit/platforms/platform_terminal.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        """Represents a platform."""
+        self._user_callback: Callable[[str], None] = None
+
+    def register_user_callback(self, callback: Callable[[str], None]) -> None:
+        """Registers a callback function to be called on user input.
+
+        Args:
+            callback: Function to call on user input
+        """
+        self._user_callback = callback
```
Comment (NoB0) [inline dialoguekit/platforms/platform_terminal.py:20]: Not sure this method is needed here as it is the same as in parent class.

File: dialoguekit/platforms/platform_terminal.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+    def listen_for_user_input(self) -> None:
+        """Listens for the user input.
+
+        Args:
+            callback: Function to call on user input
```
Comment (NoB0) [inline dialoguekit/platforms/platform_terminal.py:26]:
```suggestion
            callback: Function to call on user input.
```

File: dialoguekit/platforms/platform_terminal.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        """
+        text = input("Your response: ")
+        if self._user_callback:
+            self._user_callback(text)
+
+    def display_agent_utterance(self, utterance: Utterance) -> None:
+        """Diplays an agent utterance.
```
Comment (NoB0) [inline dialoguekit/platforms/platform_terminal.py:33]:
```suggestion
        """Displays an agent utterance.
```

File: dialoguekit/platforms/platform_terminal.py [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+        Args:
+            utterance: An instance of Utterance.
+        """
+        print(f"AGENT: {utterance.text}")
+
+    def display_user_utterance(self, utterance: Utterance) -> None:
+        """Diplays a user utterance.
```
Comment (NoB0) [inline dialoguekit/platforms/platform_terminal.py:41]:
```suggestion
        """Displays a user utterance.
```

**Review by NoB0 — APPROVED** [review]
LGTM. I made some suggestions to fix NITs in docstrings. Please check if `register_user_callback` needs to be redefined in `TerminalPlatform`.

File: dialoguekit/connector/dialogue_connector.py [hunk 11]
```diff
@@ -54,6 +55,10 @@ def dialogue_history(self):
         """Return the dialogue history."""
         return self._dialogue_history
 
+    def get_platform(self) -> Platform:
+        """Return the platform."""
```
[the remark this answers is not in the document]
Comment (IKostric) [reply dialoguekit/connector/dialogue_connector.py:59]: Done

File: dialoguekit/platforms/platform.py (hunk 1 above)
Comment (IKostric) [reply dialoguekit/platforms/platform.py:18]: Done

File: dialoguekit/platforms/platform.py (hunk 2 above)
Comment (IKostric) [reply dialoguekit/platforms/platform.py:27]: Done.

File: dialoguekit/platforms/platform.py (hunk 4 above)
Comment (IKostric) [reply dialoguekit/platforms/platform.py:42]: Done.

File: dialoguekit/platforms/platform_terminal.py (hunk 6 above)
Comment (IKostric) [reply dialoguekit/platforms/platform_terminal.py:1]: Done.

File: dialoguekit/platforms/platform_terminal.py (hunk 7 above)
Comment (IKostric) [reply dialoguekit/platforms/platform_terminal.py:20]: Removed.

File: dialoguekit/platforms/platform_terminal.py (hunk 8 above)
Comment (IKostric) [reply dialoguekit/platforms/platform_terminal.py:26]: Old

File: dialoguekit/platforms/platform_terminal.py (hunk 9 above)
Comment (IKostric) [reply dialoguekit/platforms/platform_terminal.py:33]: Done.

File: dialoguekit/platforms/platform_terminal.py (hunk 10 above)
Comment (IKostric) [reply dialoguekit/platforms/platform_terminal.py:41]: Done.

File: dialoguekit/platforms/platform.py (hunk 3 above)
Comment (IKostric) [reply dialoguekit/platforms/platform.py:24]: Done.

File: dialoguekit/platforms/platform.py (hunk 5 above)
Comment (IKostric) [reply dialoguekit/platforms/platform.py:46]: Done.

File: README.md [hunk 12]
```diff
@@ -21,70 +21,67 @@ If you want to install a DialogueKit from a specific commit or straight from git
...
+   ```
 
 ## Conventions
 
 We follow the [IAI Python Style Guide](https://github.com/iai-group/styleguide/tree/main/python).
 
 ## Contributors
 
-(Alphabetically ordered by last name) 
+(Alphabetically ordered by last name)
```
Comment (kbalog) [inline README.md:81]: Add yourself as a contributor from 2023

File: dialoguekit/platforms/platform_terminal.py [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+the terminal.
+"""
+
+from typing import Callable
+
+from dialoguekit.core.utterance import Utterance
+from dialoguekit.platforms.platform import Platform
+
+
+class TerminalPlatform(Platform):
```
Comment (kbalog) [inline dialoguekit/platforms/platform_terminal.py:13]: Rename module to `terminal_platform.py` (also update README)

**Review by kbalog — APPROVED** [review]
LGTM, just resolve the class name vs. module name mismatch and update everywhere accordingly (including README)

File: README.md (hunk 12 above)
Comment (IKostric) [reply README.md:81]: Done

**Pull request merged**